yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -1958,6 +2093,424 @@ func getSupportedVersionString(minimumVersionsFixedIn map[string]*ybversion.YBVe
+		IsPrimary    bool
+		MissingPerms []string
+		PgssEnabled  bool
+		Error        error
+	}
+
+	var results []NodePermissionResult
+
+	// Check primary
+	checkIfSchemasHaveUsagePermissions()
+	missingPerms, pgssEnabled, err := pg.GetMissingAssessMigrationPermissions()
+	if err != nil {
+		return fmt.Errorf("failed to check permissions on primary: %w", err)
+	}
+	results = append(results, NodePermissionResult{
+		NodeName:     "primary",
+		IsPrimary:    true,
+		MissingPerms: missingPerms,
+		PgssEnabled:  pgssEnabled,
+		Error:        nil,
+	})
+	pgssEnabledForAssessment = pgssEnabled // Set global flag based on primary
+
+	// Check each replica
+	for _, replica := range validatedReplicaEndpoints {
+		// Create a new Source with replica's host/port
+		replicaSource := srcdb.Source{
+			DBType:         source.DBType,
+			Host:           replica.Host,
+			Port:           replica.Port,
+			DBName:         source.DBName,
+			User:           source.User,
+			Password:       source.Password,
+			Schema:         source.Schema,
+			SSLMode:        source.SSLMode,
+			SSLCertPath:    source.SSLCertPath,
+			SSLKey:         source.SSLKey,
+			SSLRootCert:    source.SSLRootCert,
+			SSLCRL:         source.SSLCRL,
+			NumConnections: source.NumConnections,
+		}
+
+		// Create a new PostgreSQL connection for this replica
+		replicaDB := replicaSource.DB().(*srcdb.PostgreSQL)
+
+		err := replicaDB.Connect()
+		if err != nil {
+			results = append(results, NodePermissionResult{
+				NodeName:  replica.Name,
+				IsPrimary: false,
+				Error:     fmt.Errorf("failed to connect: %w", err),
+			})
+			continue
+		}
+
+		missingPerms, pgssEnabled, err := replicaDB.GetMissingAssessMigrationPermissionsForNode(true) // isReplica=true
+		replicaDB.Disconnect()
+
+		results = append(results, NodePermissionResult{
+			NodeName:     replica.Name,
+			IsPrimary:    false,
+			MissingPerms: missingPerms,
+			PgssEnabled:  pgssEnabled,
+			Error:        err,
+		})
+	}
+
+	// Analyze results and report
+	var nodesWithIssues []string
+	var nodesWithoutPgss []string
+	var nodesMissingPerms []string
+
+	utils.PrintAndLogf("\n=== Permission Check Results ===\n")
+
+	replicaCounter := 1
+	for _, result := range results {
+		// Format node display name
+		var displayName string
+		if result.IsPrimary {
+			displayName = "Primary"
+		} else {
+			displayName = fmt.Sprintf("Replica %d (%s)", replicaCounter, result.NodeName)
+			replicaCounter++
+		}
+
+		if result.Error != nil {
+			color.Red("\n%s:", displayName)
+			color.Red("  ✗ Connection failed: %v", result.Error)
+			nodesWithIssues = append(nodesWithIssues, result.NodeName)
+			continue
+		}
+
+		if len(result.MissingPerms) > 0 {
+			fmt.Printf("\n%s:\n", displayName)
+			for _, perm := range result.MissingPerms {
+				fmt.Printf("  %s %s\n", color.YellowString("⚠"), strings.TrimSpace(perm))
+			}
+			nodesMissingPerms = append(nodesMissingPerms, result.NodeName)
+			nodesWithIssues = append(nodesWithIssues, result.NodeName)
```

Comment:
I don't think you are using `nodesWithIssues`

Reply:
Removed it